R/plotting.R

```diff
@@ -296,14 +300,18 @@ plotMotifHeatmaps <- function(x,
 	.assertScalar(x = show_seqlogo, type = "logical")
 	.assertScalar(x = width.seqlogo, type = "numeric", rngExcl = c(0, Inf))
 	.assertScalar(x = use_raster, type = "logical")
+	.assertScalar(x = na_col, type = "character")
 	stopifnot(exprs = {
 	    ncol(x) == nlevels(b)
 	    all(which.plots %in% c("negLog10P", "negLog10Padj", "pearsonResid", "log2enr"))
 	    is.null(highlight) || (is.logical(highlight) && length(highlight) == nrow(x))
 	})
 	bincols <- attr(getColsByBin(b), "cols")
 	if (is.logical(cluster) && length(cluster) == 1 && cluster[1] == TRUE) {
-	    clres <- hclust(dist(assay(x, "pearsonResid")))
+	    # set NA values to 0 for the distance calculations
+	    assay <- assay(x, "pearsonResid")
+	    assay[is.na(assay)] <- 0
+	    clres <- hclust(dist(assay))
```

Comment:
I think this is probably not needed (see comment above). Also, I would not use variable names (`assay`) that are identical to function names.